bills-app/src/main/webapp/script.js

```diff
@@ -26,3 +26,19 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function addCode() {
+    getBillData();
+}
+
+function getBillData() {
+    //api key is declared in api.js as: 'key'
+    var url = 'https://api.propublica.org/congress/v1/bills/upcoming/house.json';
+    var xmlHttp = new XMLHttpRequest();
+    xmlHttp.open( "GET", url, false);
+    xmlHttp.setRequestHeader("X-API-Key", key);
+    xmlHttp.send(null);
+
+    receivedJson = xmlHttp.responseText;
```

Comment:
As discussed offline, I think you have to use onload, like here: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText I'm not sure if you can depend on responseText being ready here? And consider switching to fetch for our case, since you're already familiar with it and it might be easier to use than setting up the nitty gritty of XMLHttpRequests.